vllm/v1/core/kv_cache_manager.py

```diff
@@ -134,33 +134,26 @@ def get_computed_blocks(self,
             assert self.prefix_cache_stats is not None
             self.prefix_cache_stats.requests += 1
 
-        if len(block_hashes) * self.block_size == request.num_tokens:
-            # When prompt length is divisible by the block size and all
-            # blocks are cached, we need to recompute the last token. This
-            # have to be achieved by re-computing an entire block because
-            # allocate_slots() assumes num_computed_tokens is always a
-            # multiple of the block size. To achieve this, remove the last
-            # block hash from the block_hashes for find_longest_cache_hit
-            # This limitation can potentially be removed in the future to
-            # slightly improve the performance.
-            last_block_hash = block_hashes.pop()
-        else:
-            last_block_hash = None
-
-        computed_blocks = (
-            self.single_type_manager.find_longest_cache_hit(block_hashes))
+        # When prompt length is divisible by the block size and all
+        # blocks are cached, we need to recompute the last token. This
+        # have to be achieved by re-computing an entire block because
+        # allocate_slots() assumes num_computed_tokens is always a
+        # multiple of the block size. To achieve this, set max_cache_hit_length
+        # to prompt_length - 1 in this case.
+        # This limitation can potentially be removed in the future to
+        # slightly improve the performance.
+        max_cache_hit_length = request.num_tokens
+        if max_cache_hit_length % self.block_size == 0:
+            max_cache_hit_length -= 1
```

Comment:
I don't think we need to consider the divisibility here.

```suggestion
        # NOTE: Even if all tokens hit the cache, we must recompute the last token to obtain logits.
        # Thus, set max_cache_hit_length to prompt_length - 1. This can trigger recomputation of an entire block,
        # rather than just the single last token, because allocate_slots() requires num_computed_tokens to be 
        # block-size aligned. Removing this limitation could slightly improve performance in the future.
        max_cache_hit_length = request.num_tokens - 1
```